### PR TITLE
Open MPI Runtime Testing Harness

### PR DESCRIPTION
**Pull request by jjhursey**
The backend CI will only run if the PR to the `ompi-tests-public` repo contains changes to any file in the `runtime` directory.

Once things work well in this repo, I can activate it to watch the main Open MPI repo for changes to the submodules related to the runtime.

### CODE REVIEW

**Comment by jjhursey** [comment]
Look for the `IBM CI (ppc64le, GNU, Runtime)` status.

**Comment by jjhursey** [comment]
bot:ibm:retest

**Comment by jjhursey** [comment]
I just pushed the `runtime/bin/pretty-print-hwloc` tool which displays the hwloc binding information in a pretty print format. It uses some old code from OPAL to do so. See the README.md for details. It might be useful when creating tests.

**Comment by jjhursey** [comment]
@naughtont3 This is ready for review. Once merged then you can post a PR with changes to the `runtime` directory and bounce it off of the CI system.

File: runtime/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - **Note**: Please make your test scripts such that they can be easily run with or without the CI environment variables.
+ 2. Create a build script named `build.sh`
+    - CI will call this exactly one time (with a timeout in case it hangs).
+    - If the script returns `0` then it is considered successful. Otherwise it is considered failed.
+ 3. Create a run script named `run.sh`
+    - The script is responsible for running your test including any runtime setup/shutdown and test result inspection.
+    - CI will call this exactly one time (with a timeout in case it hangs).
+    - If the script returns `0` then it is considered successful. Otherwise it is considered failed.
+ 4. Add your directory name to the `.ci-tests` file in this directory in the order that they should be executed.
+    - Note that adding the directory is not sufficient to have CI run the test, it must be in the file.
```
Comment (naughtont3) [inline runtime/README.md:40]:
```suggestion
    - Note that adding the directory is not sufficient to have CI run the test, it must be in the `.ci-tests` file.
```

Comment (jjhursey) [reply runtime/README.md]: Thanks. Fixed.

**Comment by naughtont3** [comment]
> @naughtont3 This is ready for review. Once merged then you can post a PR with changes to the `runtime` directory and bounce it off of the CI system.

So this looks good from brief read of changeset.  I will try to give it a test to kick tires and let you know if I see anything at that point.

**Comment by jjhursey** [comment]
@naughtont3 What do you think about merging this PR, then trying it out off of a PR after it is merged. I can always push changes to either the front or backend if you find any issues.

**Comment by naughtont3** [comment]
Yep, I'd say merge and we can update there as needed.

**Pull request merged**